test/helpers.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2017 Shape Security, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function stripComments(src) {
+  // This doesn't handle html comments, because don't put html comments in your test code.
+  let out = '';
+  let lastTextBegun = 0;
+  for (let i = 0; i < src.length; ++i) {
+    let c = src[i];
+    switch (c) {
+      case '\\': {
+        // necessary for e.g. x = /\//i
+        ++i;
+        break;
+      }
+      case '"':
+      case '\'':
+      case '`': {
+        for (++i; i < src.length; ++i) {
```

Comment:
This won't work with interpolations in templates.

Reply:
Ah, right. Knew I had to be missing something.

I think I can fix it without it blowing up *too* much. Although maybe I should just patch the parser [to give us a side-table of comments](https://github.com/shapesecurity/shift-parser-js/issues/352#issuecomment-290930428)? That's probably easier at this point, actually.

Reply:
That's probably best.